src/renderer/src/routes/pages/Send/index.tsx

```diff
@@ -1,114 +1,237 @@
+  const [currentStep, setCurrentStep] = useState<number>(SendPageStep.SelectAccount)
+  const { modalNavigate } = useModalNavigate()
+  const [totalFee, setTotalFee] = useState<string>('0.00')
+  const { bsAggregatorRef } = useBsAggregatorSelector()
+  const { encryptedPasswordRef } = useEncryptedPasswordSelector()
 
-  const handleSelectAccount = account => {
+  const handleSelectAccount = (account: IAccountState) => {
     setSelectedAccount(account)
     setSelectedToken(null)
+    setSelectedAmount(0)
+    setSelectedRecipient('')
     setCurrentStep(SendPageStep.SelectToken)
   }
 
-  const handleSelectToken = token => {
+  const handleSelectToken = (token: TokenBalance) => {
     setSelectedToken(token)
+    setSelectedAmount(0)
+    setSelectedRecipient('')
     setCurrentStep(SendPageStep.SelectAmount)
   }
 
-  const handleSelectAmount = amount => {
+  const handleSelectAmount = (amount: number) => {
     setSelectedAmount(amount)
+    setSelectedRecipient('')
     setCurrentStep(SendPageStep.SelectContact)
   }
 
-  const defineStepColor = (step: SendPageStep) => {
-    if (currentStep < step) {
-      return 'text-gray-100'
+  const handleSelectRecipientAddress = (recipientAddress: string) => {
+    setSelectedRecipient(recipientAddress)
+    if (recipientAddress === '') {
+      setCurrentStep(SendPageStep.SelectContact)
+    } else {
+      setCurrentStep(SendPageStep.Send)
+    }
+  }
+
+  const populateSendService = useCallback(async (): Promise<TSendServiceResult | null> => {
+    if (!selectedAccount || !selectedToken || !selectedAmount || !selectedRecipient || !selectedAccount.encryptedKey) {
+      return null
+    }
+
+    const service = bsAggregatorRef.current.getBlockchainByName(selectedAccount.blockchain)
+
+    const key = await window.api.decryptBasedEncryptedSecret(selectedAccount.encryptedKey, encryptedPasswordRef.current)
+    if (!key) throw new Error(t('error.decryptKey'))
+    const serviceAccount = service.generateAccountFromKey(key)
+    return {
+      serviceAccount: serviceAccount,
+      service: service,
+      token: selectedToken,
+    }
+  }, [bsAggregatorRef, encryptedPasswordRef, selectedAccount, selectedAmount, selectedRecipient, selectedToken, t])
+
+  const populateTotalFee = useCallback(async () => {
+    const sendService = await populateSendService()
+    if (!sendService) {
+      return
+    }
+    if (!isCalculableFee(sendService.service)) {
+      return
+    }
+    const fee = await sendService.service.calculateTransferFee({
+      senderAccount: sendService.serviceAccount,
+      intent: {
+        receiverAddress: selectedRecipient,
+        tokenHash: sendService.token.token.hash,
+        amount: selectedAmount.toString(),
+        tokenDecimals: sendService.token.token.decimals,
+      },
+    })
+    setTotalFee(fee)
+  }, [populateSendService, selectedAmount, selectedRecipient])
+
+  useEffect(() => {
+    populateTotalFee()
+  }, [populateTotalFee])
```

Comment:
Another thing is to put the useEffects at the end of the component, as this makes it easier to read. Since the hooks are at the top, then followed by the functions, then the useEffect and then the return.